src/clj/puppetlabs/services/jruby/jruby_puppet_core.clj

```diff
@@ -61,117 +62,91 @@
+  "Searches the ruby load path for a file whose name matches that of the
+  facter jar file.  The first one found is added to the system classloader's
+  classpath.  If no match is found, an info message is written to the log
+  but no failure is returned"
+  [ruby-load-path :- (schema/both (schema/pred vector?) [schema/Str]) ]
+  (if-let [facter-jar (first
+                       (filter fs/exists?
+                               (map #(fs/file % facter-jar) ruby-load-path)))]
+    (do
+      (log/debugf "Adding facter jar to classpath from: %s" facter-jar)
+      (ks-classpath/add-classpath facter-jar))
+    (log/info "Facter jar not found in ruby load path")))
 
+(schema/defn get-initialize-pool-instance-fn :- IFn
+  [config
+   profiler :- (schema/maybe PuppetProfiler)]
+  (fn [jruby-instance]
+    (let [{:keys [ruby-load-path http-client-ssl-protocols
+                  http-client-cipher-suites
+                  http-client-connect-timeout-milliseconds
+                  http-client-idle-timeout-milliseconds
+                  use-legacy-auth-conf]} config]
+      (when-not ruby-load-path
+        (throw (Exception.
+                "JRuby service missing config value 'ruby-load-path'")))
+      (let [scripting-container (:scripting-container jruby-instance)
+            ruby-puppet-class (.runScriptlet scripting-container "Puppet::Server::Master")
+            puppet-config (jruby-internal/config->puppet-config config)
+            puppetserver-config (HashMap.)
+            env-registry (puppet-env/environment-registry)]
+        (when http-client-ssl-protocols
+          (.put puppetserver-config "ssl_protocols" (into-array String http-client-ssl-protocols)))
+        (when http-client-cipher-suites
+          (.put puppetserver-config "cipher_suites" (into-array String http-client-cipher-suites)))
+        (doto puppetserver-config
+          (.put "profiler" profiler)
+          (.put "environment_registry" env-registry)
+          (.put "http_connect_timeout_milliseconds" http-client-connect-timeout-milliseconds)
+          (.put "http_idle_timeout_milliseconds" http-client-idle-timeout-milliseconds)
+          (.put "use_legacy_auth_conf" use-legacy-auth-conf))
+        (let [jruby-puppet (.callMethodWithArgArray
+                            scripting-container
+                            ruby-puppet-class
+                            "new"
+                            (into-array Object
+                                        [puppet-config puppetserver-config])
+                            JRubyPuppet)]
+          (-> jruby-instance
+              (assoc :jruby-puppet jruby-puppet)
+              (assoc :environment-registry env-registry)))))))
+
+(schema/defn initialize-scripting-container-fn
+  [scripting-container
+   config :- jruby-schemas/JRubyConfig]
+  (doto scripting-container
+    (.setEnvironment (jruby-internal/managed-environment (jruby-internal/get-system-env) (:gem-home config)))
```

Comment:
`jruby-internal` here is still aliased to `jruby-puppet-internal`, but these functions are available in jruby-utils - I think you could use the ones from jruby-utils and delete these from this repo.